packages/cms/lib/modules/resource-form-widgets/lib/api.js

```diff
@@ -0,0 +1,180 @@
+      value: type === 'User' ? self.formatUserEmailFieldValue(item.resource, item[`confirmationEmailField${type}`]) :  item[`confirmationEmailField${type}`]
+    }
+    if (recipient) {
+      recipientData.id = recipient.id
+    }
+    await self.apos.openstadApi.addOrUpdateNotificationRecipient(recipientData);
+  }
+
+  self.disableNotificationRuleSet = async (formName) => {
+    const { ruleset } = await self.getNotificationByFormName(formName);
+    console.log('disable ', ruleset);
+    await self.apos.openstadApi.addOrUpdateNotificationRuleSet({
+      id: ruleset.id,
+      active: 0
+    });
+  }
+
+  /**
+   * Delete ruleset, template and recipient
+   * @param {string} formName
+   * @returns {Promise<void>}
+   */
+  self.deleteNotificationByFormName = async (formName) => {
+    const { ruleset, template, recipient } = await self.getNotificationByFormName(formName);
+    try {
+      const promises = [];
+      if (ruleset.id) {
+        promises.push(self.apos.openstadApi.deleteNotificationRuleSet(ruleset.id));
+      }
+      if (recipient.id) {
+        promises.push(self.apos.openstadApi.deleteNotificationRecipient(ruleset.id));
+      }
+      if (template.id) {
+        promises.push(self.apos.openstadApi.deleteNotificationTemplate(template.id));
+      }
+      await Promise.all(promises);
+    } catch(error) {
+      console.error(error);
+    }
+  }
+
+  /**
+   * Get confirmation data from the api
+   * @param {string} formName
+   * @returns {Promise<{template: object ruleset: object, recipient: object}>}
+   */
+  self.getNotificationByFormName = async (formName) => {
+    const ruleset = await self.apos.openstadApi.getNotificationRuleSetByName(formName) || {};
+    const template = ruleset.notification_template || {};
+    const recipient = ruleset.notification_recipients ? ruleset.notification_recipients[0] : {};
+
+    return {
+      ruleset,
+      template,
+      recipient,
+    }
+  }
+
+  /**
+   * Delete confirmation data when widget is removed.
+   * @param req
+   * @param doc
+   * @param options
+   * @returns {Promise<void>}
+   */
+  self.deleteConfirmationFields = async (req, doc, options) => {
+    if (doc.workflowLocale && doc.workflowLocale === 'default-draft') {
+      return;
+    }
+
+    doc.body.items.forEach((item) => {
+      const items = [].concat(...item.area1.items, ...item.area2.items, ...item.area3.items, ...item.area4.items);
+
+      items.forEach(async (item) => {
+        if (item.type === 'resource-form') {
+          // Todo: delete template, ruleset and recipient
+        }
+      });
+    });
+  }
+
+  /**
+   *
+   * @param item
+   * @returns {*[]}
+   */
+  self.getAreaItems = (item) => {
+    return [].concat(
+      ...(item.area1 && item.area1.items ? item.area1.items : []),
+      ...(item.area2 && item.area2.items ? item.area2.items : []),
+      ...(item.area3 && item.area3.items ? item.area3.items : []),
+      ...(item.area4 && item.area4.items ? item.area4.items : [])
```

Comment:
Waarom zijn er 'maar' 4 areas? Misschien handig om dat nog te vermelden in comments of in de docs